src/lumigo_tracer/utils.py

```diff
@@ -322,83 +329,17 @@ def default(self, o):
-    enforce_jsonify: bool = False,
-) -> str:
-    """
-    This function prepare the given value to send it to lumigo.
-    You should call to this function if there's a possibility that the value will be big.
-
-    Current logic:
-        Converts the data to str and if it is larger than `max_size`, we truncate it.
-
-    :param value: The value we wish to send
-    :param max_size: The maximum size of the data that we will send
-    :param enforce_jsonify: Should we raise exception in the jsonify
-    :return: The value that we will actually send
-    """
-    if isinstance(value, dict) or isinstance(value, OrderedDict):
-        try:
-            value = json.dumps(value, cls=DecimalEncoder)
-        except Exception:
-            if enforce_jsonify:
-                raise
-    elif isinstance(value, bytes):
-        try:
-            value = value.decode()
-        except Exception:
-            pass
-
-    res = str(value)
-    if len(res) > max_size:
-        return f"{res[:max_size]}...[too long]"
-    return res
-
-
+@lru_cache(maxsize=1)
 def get_omitting_regexes():
     if LUMIGO_SECRET_MASKING_REGEX in os.environ:
         given_regexes = json.loads(os.environ[LUMIGO_SECRET_MASKING_REGEX])
     elif LUMIGO_SECRET_MASKING_REGEX_BACKWARD_COMP in os.environ:
         given_regexes = json.loads(os.environ[LUMIGO_SECRET_MASKING_REGEX_BACKWARD_COMP])
     else:
         given_regexes = OMITTING_KEYS_REGEXES
-    return [re.compile(r, re.IGNORECASE) for r in given_regexes]
-
-
-def omit_keys(value: Any, regexes: Optional[List] = None):
-    """
-    This function omit problematic keys from the given value.
-    We do so in the following cases:
-    * if the value is dictionary, then we omit values by keys (recursively)
-    * if the value is a string of a json. then we parse it to dict and omit keys.
-    """
-    if not regexes:
-        regexes = get_omitting_regexes()
-    if isinstance(value, list):
-        return [omit_keys(item, regexes) for item in value]
-    if isinstance(value, (str, bytes)):
-        try:
-            # This is an optimization step. Fast identify if this is a dict
-            if value.startswith("{") if isinstance(value, str) else value.startswith(b"{"):
-                parsed_value = json.loads(value)
-                if isinstance(parsed_value, dict):
-                    return json.dumps(omit_keys(parsed_value, regexes))
-            return value
-        except Exception:
-            return value
-    if isinstance(value, dict):
-        items = {}
-        for k, v in value.items():
-            if k in SKIP_SCRUBBING_KEYS:
-                items[k] = v
-            elif isinstance(k, str) and any(r.match(k) for r in regexes):  # type: ignore
-                items[k] = "****"
-            else:
-                items[k] = omit_keys(v, regexes)
-        return items
-    return value
+    if not given_regexes:
+        return ""
+    return re.compile(fr"({'|'.join(given_regexes)})", re.IGNORECASE)
```

Comment:
create a single regex Pattern that holds all the options (lowering the `any` to c-level)